src/context/wishlist-context.js

```diff
@@ -0,0 +1,107 @@
+import { createContext, useContext, useState, useEffect } from "react";
+import { useCart } from "./cart-context";
+import { addItem } from "../services/addItem";
+import { removeItem } from "../services/removeItem";
+const WishlistContext = createContext();
+const initialState = {
+  items: [ ],
+  totalQuantity: 0,
+  error: null,
+  loading: false,
+};
+const WishlistProvider = ({ children }) => {
+  const [wishlist, setWishlist] = useState(initialState);
+  const setLoading = () =>
+    setWishlist((prev) => ({ ...prev, error: "", loading: true }));
+  const setError = (error) => {
+    setWishlist((prev) => ({ ...prev, error, loading: false }));
+  };
+  const updateWishlist = (updatedWishlist) => {
+    setWishlist((prev) => ({
+      ...prev,
+      items: [...updatedWishlist],
+      totalQuantity: [...updatedWishlist].length,
+      error: "",
+      loading: false,
+    }));
+  };
+
+  const addToWishlist = async (product) => {
+    try {
+      setLoading();
+      const response = await addItem({ source: "wishlist", product });
+      if (response.status === 201) {
+        updateWishlist(response.data.wishlist);
+      } else setError("wishlist: couldn't add item");
+    } catch (error) {
+      setError(error);
+    }
+  };
+  const removeFromWishlist = async (product) => {
+    try {
+      const response = await removeItem({
+        source: "wishlist",
+        productId: product["_id"],
+      });
+      if (response.status === 200) {
+        updateWishlist(response.data.wishlist);
+      } else setError("wishlist: Couldn't remove item");
+    } catch (error) {
+      setError(error);
+    }
+  };
+  const {addToCart} = useCart()
+  const moveToCart = (product) => {
+    addToCart(product)
+    removeFromWishlist(product)
+  }
+  const clearWishlist = () => {
+    Promise.all([...wishlist.items].map((item) => removeFromWishlist(item)))
+      .then(() => {
+        setWishlist(initialState);
+      })
+      .catch((error) => {
+        setError(error);
+      });
+  };
+
+  useEffect(() => {
+    const token =
+      localStorage.getItem("token") ?? sessionStorage.getItem("token") ?? "";
```

Comment:
Since it's your app, you would know where you are storing the token, is it local storage or session storage, so here you should not blindly check for both.

Reply:
There is a remember me feature which stores the token in the localStorage else it'll store the token in session storage